d2/src/main/java/com/linkedin/d2/balancer/D2ClientBuilder.java

```diff
@@ -643,6 +645,11 @@ public D2ClientBuilder setDualReadStateManager(DualReadStateManager dualReadStat
     return this;
   }
 
+  public D2ClientBuilder setLoadBalancerThreadPool(ThreadPoolExecutor loadBalancerThreadPool) {
```

Comment:
It's better to make this a more generic signature here, otherwise it limits the threadpool implementations that can be used. In this case the minimum interface you need is `ExecutorService`, which is the basic interface for a threapool in Java.

Reply:
Make sense. Update it to more generic signature :
`ExecutorService`